h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostExtension.java

```diff
@@ -75,6 +76,19 @@ public static NativeLibraryLoaderChain getLoader() throws IOException {
     return(NativeLibraryLoaderChain) loader;
   }
 
+  @Override
+  public void onLocalNodeStarted() {
+    if (!isEnabled())
+      return;
+    if (H2O.ARGS.off_heap_memory_ratio > 0) {
+      MemoryCheck.Report report = MemoryCheck.runCheck(H2O.ARGS.off_heap_memory_ratio);
+      if (!report.isOffHeapRequirementMet()) {
+        LOG.warn("There doesn't seem to be enough memory available for XGBoost model training, " +
+                "training XGBoost models is not advised. Details: " + report);
```

Comment:
Shouldn't you mention the off-heap memory ratio in the message? So that the users knows what to change?
Also wondering how we could raise those warnings to the client on startup, but it's for another PR.

Reply:
Added the ratio information to the message.

Yes, eventually we should check the preconditions before the algo is started. We could do it in a similar way we check for XGBoost availability.